aten/src/ATen/native/mps/operations/Copy.mm

```diff
@@ -436,6 +459,14 @@ void copy_blit_mps(void* dst, const void* src, size_t size) {
   } else {
     src = src_;
   }
+
+  // Scatter to `dst` if the memory is not contiguous
+  // If the memory is not contiguous, it means that the tensor has strides and we would not be
+  // able to do the copy using a single blit
+  if (!dst_.is_contiguous()) {
```

Comment:
Why in this specific copy variant we call scatter and in others we call gather? What is different in this specific copy variant?

Reply:
Right that makes sense, but the check, should have alias Info as a condition . This doesn't look complete.